src/GenericFactory.sol

```diff
@@ -62,7 +62,6 @@ contract GenericFactory is IGenericFactory, Ownable
         require(aTokenA != address(0), "FACTORY: ZERO_ADDRESS");
         require(getPair[aTokenA][aTokenB][aCurveId] == address(0), "FACTORY: PAIR_EXISTS");
         address lCodePointer = _getByteCode[aCurveId];
-        require(lCodePointer != address(0), "FACTORY: INVALID_CURVE_ID");
 
```

Comment:
removed this line because if `aCurveId` was invalid, the caller would get an index out of bound error and would not reach the `require` statement. Also added a corresponding test case for that